src/tv2-common/actions/CoreActionExecutionContext.ts

```diff
@@ -0,0 +1,238 @@
+		return this.core.getPieceInstances(part) as Promise<Array<IBlueprintPieceInstance<PieceMetaData>>>
+	}
+
+	public async getResolvedPieceInstances(
+		part: 'current' | 'next'
+	): Promise<Array<IBlueprintResolvedPieceInstance<PieceMetaData>>> {
+		return this.core.getResolvedPieceInstances(part) as Promise<Array<IBlueprintResolvedPieceInstance<PieceMetaData>>>
+	}
+
+	public async findLastPieceOnLayer(
+		sourceLayerId: string | string[],
+		options?: {
+			excludeCurrentPart?: boolean
+			originalOnly?: boolean
+			pieceMetaDataFilter?: any
+		}
+	): Promise<IBlueprintPieceInstance<PieceMetaData> | undefined> {
+		return this.core.findLastPieceOnLayer(sourceLayerId, options) as Promise<
+			IBlueprintPieceInstance<PieceMetaData> | undefined
+		>
+	}
+
+	public async findLastScriptedPieceOnLayer(
+		sourceLayerId: string | string[],
+		options?: { excludeCurrentPart?: boolean; pieceMetaDataFilter?: any }
+	): Promise<IBlueprintPiece<PieceMetaData> | undefined> {
+		return this.core.findLastScriptedPieceOnLayer(sourceLayerId, options) as Promise<
+			IBlueprintPiece<PieceMetaData> | undefined
+		>
+	}
+
+	public async getPartInstanceForPreviousPiece(
+		piece: IBlueprintPieceInstance<PieceMetaData>
+	): Promise<IBlueprintPartInstance<unknown>> {
+		return this.core.getPartInstanceForPreviousPiece(piece)
+	}
+
+	public async getPartForPreviousPiece(
+		piece: IBlueprintPieceDB<PieceMetaData>
+	): Promise<IBlueprintPart<unknown> | undefined> {
+		return this.core.getPartForPreviousPiece(piece)
+	}
+
+	public async stopPiecesOnLayers(sourceLayerIds: string[], timeOffset?: number): Promise<string[]> {
+		return this.core.stopPiecesOnLayers(sourceLayerIds, timeOffset)
+	}
+
+	public async stopPieceInstances(pieceInstanceIds: string[], timeOffset?: number): Promise<string[]> {
+		return this.core.stopPieceInstances(pieceInstanceIds, timeOffset)
+	}
+
+	public async takeAfterExecuteAction(take: boolean): Promise<boolean> {
+		return this.core.takeAfterExecuteAction(take)
+	}
+
+	public notifyUserError(message: string, params?: { [key: string]: any }): void {
+		return this.core.notifyUserError(message, params)
+	}
+
+	public notifyUserWarning(message: string, params?: { [key: string]: any }): void {
+		return this.core.notifyUserWarning(message, params)
+	}
+
+	public getHashId(originString: string, originIsNotUnique?: boolean | undefined) {
+		return this.core.getHashId(originString, originIsNotUnique)
+	}
+
+	public unhashId(hash: string) {
+		return this.core.unhashId(hash)
+	}
+
+	public logDebug(message: string) {
+		return this.core.logDebug(message)
+	}
+
+	public logInfo(message: string) {
+		return this.core.logInfo(message)
+	}
+
+	public logWarning(message: string) {
+		return this.core.logWarning(message)
+	}
+
+	public logError(message: string) {
+		return this.core.logError(message)
+	}
+
+	public getShowStyleConfig(): unknown {
```

Comment:
This is the type core returns because it can't know the type of the data that the blueprints store there after preprocessing, and this class is the only wrapper around core context done this way, all the other will still return `unknown` which the `ExtendedShowStyleContextImpl` still has to deal with, so I don't think it is reasonable to change this particular one